src/main/java/forestry/apiculture/ApiaristAI.java

```diff
@@ -0,0 +1,171 @@
+/*******************************************************************************
+ * Copyright (c) 2011-2014 SirSengir.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Lesser Public License v3
+ * which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/lgpl-3.0.txt
+ *
+ * Various Contributors including, but not limited to:
+ * SirSengir (original work), CovertJaguar, Player, Binnie, MysteriousAges
+ ******************************************************************************/
+
+package forestry.apiculture;
+
+import forestry.api.apiculture.EnumBeeType;
+import forestry.apiculture.blocks.BlockApiculture;
+import forestry.apiculture.items.ItemBeeGE;
+import forestry.apiculture.tiles.TileBeeHouse;
+import forestry.core.tiles.TileUtil;
+import forestry.core.utils.InventoryUtil;
+
+import net.minecraft.block.Block;
+import net.minecraft.entity.ai.EntityAIMoveToBlock;
+import net.minecraft.entity.passive.EntityVillager;
+import net.minecraft.inventory.InventoryBasic;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public class ApiaristAI extends EntityAIMoveToBlock {
+	private final EntityVillager villager;
+	private boolean hasDrone;
+	private boolean hasPrincess;
+	private InventoryBasic villagerInventory;
+
+	private static final int SLOT_PRODUCT_1 = InventoryBeeHousing.SLOT_PRODUCT_1;
+	private static final int SLOT_PRODUCT_COUNT = InventoryBeeHousing.SLOT_PRODUCT_COUNT;
+	private static final int SLOT_QUEEN = InventoryBeeHousing.SLOT_QUEEN;
+	private static final int SLOT_DRONE = InventoryBeeHousing.SLOT_DRONE;
+
+	public ApiaristAI(EntityVillager villager, double speed) {
+		super(villager, speed, 16);
+		this.villager = villager;
+		villagerInventory = villager.getVillagerInventory();
+	}
+
+	@Override
+	public boolean shouldExecute() {
+		if (this.runDelay < 0) {
```

Comment:
`this.runDelay` is never less than 0 so this is never true.
Please test this more carefully.